tests/Admin/AdminTest.php

```diff
@@ -2495,6 +2510,37 @@ public function testAdminWithoutControllerName(): void
         $this->assertNull($admin->getBaseControllerName());
     }
 
+    public function testAdminAvoidInifiniteLoop(): void
+    {
+        $this->expectNotToPerformAssertions();
+
+        $formFactory = new FormFactory(new FormRegistry([], new ResolvedFormTypeFactory()));
+
+        $admin = new AvoidInfiniteLoopAdmin('code', \stdClass::class, null);
+        $admin->setSubject(new \stdClass());
+
+        $admin->setFormContractor(new FormContractor($formFactory));
+
+        $admin->setShowBuilder(new ShowBuilder());
+
+        $admin->setListBuilder(new ListBuilder());
+
+        $pager = $this->createMock(PagerInterface::class);
+        $admin->setDatagridBuilder(new DatagridBuilder($formFactory, $pager));
+
+        $validator = $this->createMock(ValidatorInterface::class);
+        $validator->method('getMetadataFor')->willReturn($this->createMock(MemberMetadata::class));
+        $admin->setValidator($validator);
+
+        $routeGenerator = $this->getMockForAbstractClass(RouteGeneratorInterface::class);
```

Comment:
You are already using `createMock()` for interfaces. Why is this case different?